dask/order.py

```diff
@@ -381,6 +381,75 @@ def process_runnables() -> None:
     # writing, the most expensive part of ordering is the prep work (mostly
     # connected roots + sort_key) which can be reused for multiple orderings.
 
+    # Only build this mapping if it's needed. This can be expensive for
+    # all-to-all communications, but we normally don't need it in this
+    # case
+    occurences_grouped_sorted: dict[int, list[Key]] = {}
+
+    def _built_occurrences_group_sorted() -> None:
+        if occurences_grouped_sorted:
+            return
+        occurrences: defaultdict[Key, int] = defaultdict(int)
+        for t in leaf_nodes:
+            for r in roots_connected[t]:
+                occurrences[r] += 1
+        occurences_grouped = defaultdict(set)
+        for root, occ in occurrences.items():
+            occurences_grouped[occ].add(root)
+        for k, v in occurences_grouped.items():
+            occurences_grouped_sorted[k] = sorted(v, key=sort_key, reverse=True)
```

Comment:
Unrelated to your changes but given performance here has been an issue...

We should be able to use the knowledge that all `v` are root nodes and perform the sort on all roots once.
With this knowledge we should be able to do sort every iteration in N instead of N logN

Reply:
this is gone now